framework/src/meshlet_helpers.cpp

```diff
@@ -0,0 +1,139 @@
+﻿#include <gvk.hpp>
+
+namespace gvk
+{
+	std::vector<meshlet> divide_into_meshlets(const std::vector<std::tuple<avk::resource_ownership<model_t>, std::vector<mesh_index_t>>>& aModels,
+		const uint32_t aMaxVertices, const uint32_t aMaxIndices, const bool aCombineSubmeshes)
+	{
+		return divide_into_meshlets(aModels, divide_into_very_bad_meshlets, aMaxVertices, aMaxIndices, aCombineSubmeshes);
+	}
+
+	std::vector<meshlet> divide_into_meshlets_cached(gvk::serializer& aSerializer,
+		const std::vector<std::tuple<avk::resource_ownership<gvk::model_t>, std::vector<mesh_index_t>>>& aModels,
+		const uint32_t aMaxVertices, const uint32_t aMaxIndices, const bool aCombineSubmeshes)
+	{
+		return divide_into_meshlets_cached(aSerializer, aModels, divide_into_very_bad_meshlets, aMaxVertices, aMaxIndices, aCombineSubmeshes);
+	}
+
+	template <typename F>
+	std::vector<meshlet> divide_into_meshlets(const std::vector<std::tuple<avk::resource_ownership<gvk::model_t>, std::vector<mesh_index_t>>>& aModels, F&& aMeshletDivision,
+		const uint32_t aMaxVertices, const uint32_t aMaxIndices, const bool aCombineSubmeshes)
+	{
+		std::vector<meshlet> meshlets;
+		for (auto& pair : aModels) {
+			auto model = std::get<avk::resource_ownership<model_t>>(pair);
```

Comment:
I think, this is peforming a copy assignment, not a move assignment, which would only work if shared ownership was enabled on the model. We only need the reference to the `avk::resource_ownership` for now, however.

```suggestion
			auto& model = std::get<avk::resource_ownership<model_t>>(pair);
```